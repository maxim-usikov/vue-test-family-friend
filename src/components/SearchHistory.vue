<template>
  <section>
    <a
      href="#"
      @click="toggleShow"
    >История запросов:</a>
    <HistoryEmptyList v-if="show && items.length === 0" />
    <HistoryList v-else-if="show">
      <HistoryListItem
        v-for="item in items"
        :key="item.id"
        v-bind="item"
      />
    </HistoryList>
  </section>
</template>

<script>
import HistoryEmptyList from '@/components/HistoryEmptyList.vue';
import HistoryList from '@/components/HistoryList.vue';
import HistoryListItem from '@/components/HistoryListItem.vue';

export default {
  name: 'SearchHistory',

  components: {
    HistoryEmptyList,
    HistoryList,
    HistoryListItem,
  },

  props: {
    items: {
      type: Array,
      required: true,
    },
  },

  data() {
    return {
      show: false,
    };
  },

  methods: {
    toggleShow() {
      this.show = !this.show;
    },
  },
};
</script>
