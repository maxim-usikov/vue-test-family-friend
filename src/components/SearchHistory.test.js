import { shallowMount } from '@vue/test-utils';

import SearchHistory from './SearchHistory.vue';

describe('SearchHistory', () => {
  let $c;

  beforeEach(() => {
    $c = shallowMount(SearchHistory, {
      propsData: {
        items: [],
      },
    });
  });

  it('is a Vue instance', () => {
    expect($c.isVueInstance()).toBeTruthy();
  });

  it('name is `SearchHistory`', () => {
    expect($c.name()).toBe('SearchHistory');
  });

  it('renders correctly', () => {
    expect($c.element).toMatchSnapshot();
  });

  it('toggles show', () => {
    const a = $c.find('a');

    expect($c.vm.show).toBeFalsy();
    a.trigger('click');
    expect($c.vm.show).toBeTruthy();
    a.trigger('click');
    expect($c.vm.show).toBeFalsy();
  });

  it('renders HistoryEmptyList component', () => {
    $c.setProps({ items: [] });
    $c.setData({ show: true });

    expect($c.element).toMatchSnapshot();
  });

  it('renders HistoryList component', () => {
    $c.setProps({
      items: [
        {
          id: 111,
          query: 'test',
          createdAt: new Date('December 26, 1990 00:00:00'),
          response: { data: { search: { repositoryCount: 888 } } },
        },
        {
          id: 222,
          query: 'vue',
          createdAt: new Date('December 26, 1990 20:00:00'),
          response: { data: { search: { repositoryCount: 999 } } },
        },
      ],
    });
    $c.setData({ show: true });

    expect($c.element).toMatchSnapshot();
  });
});
