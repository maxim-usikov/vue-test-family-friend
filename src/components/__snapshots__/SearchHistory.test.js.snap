// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchHistory renders HistoryEmptyList component 1`] = `
<div>
  <a
    href="#"
  >
    История запросов:
  </a>
   
  <historyemptylist-stub />
</div>
`;

exports[`SearchHistory renders HistoryList component 1`] = `
<div>
  <a
    href="#"
  >
    История запросов:
  </a>
   
  <historylist-stub>
    <historylistitem-stub
      createdat="Wed Dec 26 1990 00:00:00 GMT+0300 (Moscow Standard Time)"
      id="111"
      query="test"
      response="[object Object]"
    />
    <historylistitem-stub
      createdat="Wed Dec 26 1990 20:00:00 GMT+0300 (Moscow Standard Time)"
      id="222"
      query="vue"
      response="[object Object]"
    />
  </historylist-stub>
</div>
`;

exports[`SearchHistory renders correctly 1`] = `
<div>
  <a
    href="#"
  >
    История запросов:
  </a>
   
  <!---->
</div>
`;
